docs/src/ci.md

```diff
@@ -0,0 +1,143 @@
+
+- **Stage: Integration Tests**: Execute the unit tests of `orig` via `Pkg.test()`.
+- **Stage: Generate integration Tests**: Download the `integTestGen.jl` script from the `QED.jl` package via `git clone`. The script is executed with the name of the subpackage `orig`. The script traverses the dependency tree of the `QED.jl` and it is searching for sub-packages which has a dependency of package `orig`. For each package with the dependency of package `orig`, the generator generates a job yaml. By default, it uses the upstream repository and development branch of the package tested in the integration test. The repository and commit can be changed via the environment parameter in the git commit message. The `integTestGen.jl` uses the `Package.toml` of the `QED.jl` package because all sub-packages are direct and indirect dependencies of it.
+- **Stage: Run Integration Tests**: This stage uses the generated job yaml to create and run new test jobs. It uses the [GitLab CI child pipeline](https://about.gitlab.com/blog/2020/04/24/parent-child-pipelines/#dynamically-generating-pipelines) mechanism.
+- **Stage: Integration Tests of Sub package N**: Each job clones the repository of the subpackage. After the clone, it uses the Julia function `Pkg.develop(path="$CI_package_DIR")` to replace the dependency to the package `orig` with the modified version of the Pull Request and execute the tests of the subpackage via `Pkg.test()`.
+
+!!! note
+
+    If a sub package triggers an integration test, the main package `QED.jl` is passive. It does not get any notification or trigger any script. The repository is simply cloned.
+
+The commit message is defined in the environment variable `CI_COMMIT_MESSAGE` by GitLab CI. If the variable is not defined, the script ignores the commit message. If you want to disable reading the commit message you can set the name of the commit message variable to a undefined variable via first argument of the `integTestGen.jl` script. We use this, when we execute the CI on the `main` or `dev` branch. On this branches, it should be not possible to use custom URLs for unit or integration tests. Therefore we disable it, which also allows to use `CI_INTG_PKG_URL_` variables as regular part of the merge commit message.
+
+## running locally
+
+The `integTestGen.jl` script has a special behavior. It creates is own `Project.toml` in a temporary folder and switches it's project environment to it. Therefor you need to start the script with the project path `--project=ci/integTestGen`. You need also to set two environment variables:
+
+- **CI_DEPENDENCY_NAME:** Name of the package, where to run the integration tests. For example `QEDbase`.
```

Comment:
Not sure what you mean by `where to run`. It's not a directory like the next one, is it?